### PR TITLE
update to kim-api-2.1.1 and openkim-models-2019-07-25

### PR DESCRIPTION
**Pull request by ellio167**

### CODE REVIEW

**Comment by ellio167** [comment]
@adamjstewart OK, I've removed them! Thanks.

File: var/spack/repos/builtin/packages/openkim-models/package.py [hunk 1]
```diff
@@ -19,19 +19,23 @@ class OpenkimModels(CMakePackage):
...
                      + '/lib/kim-api/model-drivers').format(prefix))
-        args.append('-DKIM_API_MODEL_INSTALL_PREFIX={0}/lib/kim-api/models'
-                    .format(prefix))
-        args.append('-DKIM_API_SIMULATOR_MODEL_INSTALL_PREFIX={0}/lib/'
-                    'kim-api/simulator-models'
-                    .format(prefix))
+        args.append(('-DKIM_API_PORTABLE_MODEL_INSTALL_PREFIX={0}'
+                     + '/lib/kim-api/portable-models').format(prefix))
+        args.append(('-DKIM_API_SIMULATOR_MODEL_INSTALL_PREFIX={0}'
+                     + '/lib/kim-api/simulator-models').format(prefix))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openkim-models/package.py]: You changed the arg names and directories. Does this work for both `2019-03-29` _and_ `2019-07-25`? If not, you'll have to use different args for different versions.

Comment (ellio167) [reply var/spack/repos/builtin/packages/openkim-models/package.py]: Ah Ha!  Actually this depends on the kim-api package version.  How do I go about setting these args depending on the version of its dependency?  (Thanks for your help!)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/openkim-models/package.py]: You can do something like:
```python
if self.spec.satisfies('@1.2.3:'):
    ...
else:
    ...
```

Comment (ellio167) [reply var/spack/repos/builtin/packages/openkim-models/package.py]: Well....  I've made it hard for myself...
How can I limit the versions of a dependency that are compatible with a specific version of the package?

In words, I want to specify: "versions `openkim-models@2019-07-25:` must be used with version `kim-api@2.1.0:` " and "version `openkim-models@2019-03-29` must be used with version `kim-api@:2.0.2` "

Is that possible?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/openkim-models/package.py]:
```python
depends_on('kim-api@2.1.0:', when='@2019-07-25:')
depends_on('kim-api@:2.0.2', when='@2019-03-29')
```

**Comment by ellio167** [comment]
@adamjstewart Thanks for our help.

**Comment by adamjstewart** [comment]
By the way, if you add the following lines to your packages:
```python
maintainers = ['ellio167']
```
we will know to ping you if anyone reports a bug with your packages or wants to make changes to your packages in Spack.

**Comment by adamjstewart** [comment]
Flake8 is failing:
```
var/spack/repos/builtin/packages/openkim-models/package.py:41: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/openkim-models/package.py:44: [E111] indentation is not a multiple of four
var/spack/repos/builtin/packages/openkim-models/package.py:48: [E303] too many blank lines (2)
```

**Pull request merged**